error/s2n_errno.c

```diff
@@ -125,3 +125,24 @@ const char *s2n_strerror(int error, const char *lang)
 
     return no_such_error;
 }
+
+int s2n_error_get_type(int error)
+{
+    if (error < S2N_ERR_T_IO_START) {
+        return S2N_ERR_T_OK;
+    } else if (error < S2N_ERR_T_CLOSED_START) {
+        return S2N_ERR_T_IO;
+    } else if (error < S2N_ERR_T_BLOCKED_START) {
+        return S2N_ERR_T_CLOSED;
+    } else if (error < S2N_ERR_T_ALERT_START) {
+        return S2N_ERR_T_BLOCKED;
+    } else if (error < S2N_ERR_T_PROTO_START) {
+        return S2N_ERR_T_ALERT;
+    } else if (error < S2N_ERR_T_INTERNAL_START) {
+        return S2N_ERR_T_PROTO;
+    } else if (error < S2N_ERR_T_USAGE_START) {
+        return S2N_ERR_T_INTERNAL;
+    } else {
+        return S2N_ERR_T_USAGE;
+    }
+}
```

Comment:
One trick you could do here (I'll leave it up to you) is to use powers of two. For example, let's say we make space for 32 error types. We know that int is at least 32-bits on the platforms we support, so we could say that the error codes between [0 - 67108863] are the first error type, then [67108864 - 134217727] are the 2nd and so on, in blocks of 2 ** 26. 

Then the lookup can be an O(1) table, e.g. 

int types[32] = { S2N_ERR_T_OK, S2N_ERR_T_IO, ... S2N_ERR_T_USAGE };

return types[error >> 26];